syncd/FlexCounter.cpp

```diff
@@ -3433,3 +3519,110 @@ void FlexCounter::notifyPoll()
     m_readyToPoll = true;
     m_pollCond.notify_all();
 }
+
+bool FlexCounter::isOnGearbox()
+{
+    SWSS_LOG_ENTER();
+
+    return m_dbCounters == "GB_COUNTERS_DB";
+}
+
+bool FlexCounter::isGearboxEnabled()
+{
+    SWSS_LOG_ENTER();
+
+    return access("/usr/share/sonic/hwsku/gearbox_config.json", F_OK) != -1;
```

Comment:
this is also not good to hardcode config path :( please don't use hardcoded config location to figure out whether this is gearbox or not

isOnGearbox and isGearboxEnabled basically are the same? why there could not be only 1 function for this? do you need some special action if gearbox is enabled but non on gearbox? seems like just putting that config would have some impact even if syncd will not run in gearbox

i have a proposal here: can you have 1 flag maybe as commandline option or an extra option in context_config.json that specifies this is gearbox ? and then syncd would be able to figure this out based on that parameter rather hardcoding db names or config location

Reply:
Have a new design https://github.com/Azure/sonic-swss-common/pull/622. In application (counter reader) side, it handles the logic of counter combining of asic port and gearbox port. Then we do not need to change anything in syncd/gbsyncd, which only simply collects counters from asic/phy chip.

Reply:
ok, then after fixing this in common repo, please close this pr

Reply:
Sure, will do it.